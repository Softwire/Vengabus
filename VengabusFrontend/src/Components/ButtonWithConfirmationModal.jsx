import React from 'react';
import { Modal, Alert, Button, Tooltip, OverlayTrigger } from 'react-bootstrap';

const defaultState = {
    show: false
};

class ButtonWithConfirmationModal extends React.Component {
    constructor(props) {
        super(props);

        this.state = defaultState;
    }

    handleOpen = () => {
        const afterOpen = (this.props.afterShowModalAction || (() => { }));
        this.setState({ show: true }, afterOpen);
    }

    handleConfirm = () => {
        this.props.confirmAction();
        this.handleClose();
    }

    handleClose = () => {
        const afterClose = (this.props.afterCloseModalAction || (() => { }));
        this.setState(defaultState, afterClose);
    }

    render() {
        const style = this.props.buttonDisabled ? {
            pointerEvents: 'none'
        } : {};

        const tooltip = (
            <Tooltip id="tooltip">
                <strong>{this.props.tooltipMessage}</strong>
            </Tooltip>
        );

        const button =
            <Button
                onClick={this.handleOpen}
                bsStyle={this.props.buttonStyle || "danger"}
                disabled={!!this.props.buttonDisabled}
                style={style}
            >
                {this.props.buttonText}
            </Button>;

        const buttonWithToolTip =
            <OverlayTrigger rootClose overlay={tooltip} placement="top">
                <div style={{ display: 'inline-block', cursor: 'not-allowed', float: 'right' }}>
                    {button}
                </div>
            </OverlayTrigger>;


        return (
            <React.Fragment>
                {this.props.tooltipMessage ? buttonWithToolTip : button}
                <Modal show={this.state.show} onHide={this.handleClose} >
                    <Modal.Header>
                        <Modal.Title>{this.props.modalTitle}</Modal.Title>
                    </Modal.Header>
                    <Modal.Body>
                        {this.props.modalBody ?
                            <Alert bsStyle={this.props.buttonStyle ? this.props.buttonStyle : "danger"}>
                                {this.props.modalBody}
                            </Alert > : null
                        }
                        {this.props.children}
                    </Modal.Body>
                    <Modal.Footer>
                        <Button id="confirm" onClick={this.handleConfirm} bsStyle="danger">{this.props.confirmButtonText}</Button>
                        <Button id="cancel" onClick={this.handleClose}>{this.props.cancelButtonText || "Cancel"}</Button>
                    </Modal.Footer>
                </Modal>
            </React.Fragment >
        );
    }
}

export { ButtonWithConfirmationModal };
