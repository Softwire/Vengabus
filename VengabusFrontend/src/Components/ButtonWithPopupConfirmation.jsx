import React from 'react';
import { Modal, Alert, Button } from 'react-bootstrap';

const defaultState = {
    show: false
};

class ButtonWithPopupConfirmation extends React.Component {
    constructor(props) {
        super(props);

        this.state = defaultState;
    }

    handleConfirm = () => {
        this.props.confirmAction();
        this.handleClose();
    }

    handleClose = () => {
        this.props.closeModalAction();
        this.setState(defaultState);
    }

    handleOpening = () => {
        this.setState({ show: true });
        this.props.showModalAction();
    }

    render() {
        return (
            <React.Fragment>
                <Button onClick={this.handleOpening} bsStyle={this.props.buttonStyle}>
                    {this.props.buttonText}
                </Button>
                <Modal show={this.state.show} onHide={this.handleClose} >
                    <Modal.Header>
                        <Modal.Title>{this.props.modalTitle}</Modal.Title>
                    </Modal.Header>

                    <Modal.Body>
                        <Alert bsStyle="danger">
                            {this.props.modalBody}
                        </Alert >
                        {this.props.children}
                    </Modal.Body>
                    <Modal.Footer>
                        <Button id="confirm" onClick={this.handleConfirm} bsStyle="danger">{this.props.confirmButtonText}</Button>
                        <Button id="cancel" onClick={this.handleClose}>{this.props.cancelButtonText}</Button>
                    </Modal.Footer>
                </Modal>
            </React.Fragment>
        );
    }
}

export { ButtonWithPopupConfirmation };
