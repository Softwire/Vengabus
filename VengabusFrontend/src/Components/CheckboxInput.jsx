import React, { Component } from 'react';
import { css } from 'emotion';
import { Checkbox, Glyphicon, OverlayTrigger } from 'react-bootstrap';

/**
 * @prop {bool} data The data that is set by the checkbox.
 * @prop {node} tooltip The tooltip element that is displayed when hovering over the info glyphicon. If undefined then no info tooltip.
 * @prop {function} onChange The function that is called when the value of the checkbox is changed.
 * @prop {string} text The name of the property that is being set by the checkbox.
 */
export class CheckboxInput extends Component {

    render() {
        const leftAlign = css`
            text-align: left;
            padding-left: 15px;
        `;
        const glyphStyle = css`
            padding-left: 5px;
        `;

        return (
            <div>
                {
                    this.props.tooltip ? (
                        <p className={leftAlign}>
                            {this.props.text}
                            <OverlayTrigger placement="right" overlay={this.props.tooltip}>
                                <Glyphicon glyph="info-sign" className={glyphStyle} />
                            </OverlayTrigger>
                        </p>
                    ) : (
                            <p className={leftAlign}>{this.props.text}</p>
                        )
                }
                <Checkbox
                    className={leftAlign}
                    checked={this.props.data}
                    onChange={(event) => {
                        this.props.onChange(event.target.checked);
                    }}
                />
            </div>
        );
    }

}


