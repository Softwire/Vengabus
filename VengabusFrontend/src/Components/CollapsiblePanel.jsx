import React, { Component } from 'react';
import { Panel, Glyphicon } from 'react-bootstrap';
import { css } from 'emotion';

export class CollapsiblePanel extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            glyphicon: "plus"
        };
    }


    toggleGlyphicon = () => {
        this.setState({
            glyphicon: this.state.glyphicon === "plus" ? "minus" : "plus"
        });
    }

    render() {
        const panelStyle = css`
            .panel {
                margin-bottom: 10px;
            }
            .panel-heading {
                padding: 0 15px;
                color: #333;
                background-color: #f5f5f5;
                border-color: #ddd;
            }
            .panel-body { 
                padding: 0;
            }
            pre {
                background-color: #ffffff;
                border: none;
                margin: 0;
            }
            pre p {
                margin: 0;
            }
            
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const glyphiconStyle = css`
            .glyphicon-plus {
                margin-left: 1px;
                margin-right: 10px;
            }
            .glyphicon-minus {
                margin-right: 11px;
            }
        `;
        return (
            <Panel defaultExpanded={this.props.isDefaultExpanded} className={panelStyle} onToggle={() => this.toggleGlyphicon()} >
                <Panel.Toggle>
                    <Panel.Heading>
                        <Panel.Title className={glyphiconStyle} >
                            <Glyphicon glyph={this.state.glyphicon} /><h4 className={headerStyle}>{this.props.panelTitle}</h4>
                        </Panel.Title>
                    </Panel.Heading>
                </Panel.Toggle>
                <Panel.Collapse>
                    <Panel.Body>{this.props.panelContent}</Panel.Body>
                </Panel.Collapse>
            </Panel >

        );
    }

}