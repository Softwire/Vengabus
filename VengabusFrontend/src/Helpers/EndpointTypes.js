var EndpointTypes = Object.freeze({
    QUEUE: "queue",
    TOPIC: "topic",
    SUBSCRIPTION: "subscription",
    MESSAGE: "message",
    DEADLETTER: "deadletters"
});


var typeToTitle = (type) => {
    switch (type) {
        case EndpointTypes.MESSAGE:
            return "Messages";
        case EndpointTypes.SUBSCRIPTION:
            return "Subscriptions";
        case EndpointTypes.QUEUE:
            return "Queues";
        case EndpointTypes.TOPIC:
            return "Topics";
        case EndpointTypes.DEADLETTER:
            return "Deadletters";
        default:
            throw new Error('Invalid endpoint type.');
    }
}
export { EndpointTypes, typeToTitle };