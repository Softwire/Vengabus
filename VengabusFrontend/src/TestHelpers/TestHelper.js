import { Button } from "react-bootstrap";

class TestHelper {
    /*
     * It returns a promise that waits until the state of the components is updated and then continues.
     * 
     * Expected usage is: 
     *     return testHelper.afterReactHasUpdated().then(() => {
     *         expect('actualValue').toEqual('expectedValue');
     *     });
     * 
     * Notes on multi-layered async tests (e.g. wait for this, and then wait for that, and then expect a result) will be forthcoming.
     */
    afterReactHasUpdated(noArgsGiven) {
        if (noArgsGiven) {
            throw new Error('afterReactHasUpdated() does not expect to take any arguments. See expected usage for further details.');
        }

        return new Promise(function (resolve) {
            setTimeout(resolve, 0);
        });
    }
    clickElementWithId(wrapper, id) {
        wrapper.find(id).at(0).simulate("click");
    }
}

export const testHelper = new TestHelper();
