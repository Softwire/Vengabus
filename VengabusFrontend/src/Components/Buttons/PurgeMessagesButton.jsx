import React from 'react';
import { Glyphicon } from 'react-bootstrap';
import { serviceBusConnection } from '../../AzureWrappers/ServiceBusConnection';
import { EndpointTypes } from '../../Helpers/EndpointTypes';
import Lodash from 'lodash';
import { ButtonWithConfirmationModalAndNotification } from './ButtonWithConfirmationModalAndNotification';

class PurgeMessagesButton extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            modalBody: ""
        };
    }

    getOnPurgeConfirmedHandler = () => {
        if (this.props.messageType === EndpointTypes.MESSAGE) {
            switch (this.props.type) {
                case EndpointTypes.TOPIC:
                    return () => this.vengaServiceBusService.purgeTopicMessages(this.props.endpointName);
                case EndpointTypes.QUEUE:
                    return () => this.vengaServiceBusService.purgeQueueMessages(this.props.endpointName);
                case EndpointTypes.SUBSCRIPTION:
                    return () => this.vengaServiceBusService.purgeSubscriptionMessages(this.props.parentName, this.props.endpointName);
                default: break;
            }
        } else {
            switch (this.props.type) {
                case EndpointTypes.TOPIC:
                    return () => this.vengaServiceBusService.purgeTopicDeadletterMessages(this.props.endpointName);
                case EndpointTypes.QUEUE:
                    return () => this.vengaServiceBusService.purgeQueueDeadletterMessages(this.props.endpointName);
                case EndpointTypes.SUBSCRIPTION:
                    return () => this.vengaServiceBusService.purgeSubscriptionDeadletterMessages(this.props.parentName, this.props.endpointName);
                default: break;
            }
        }
    }

    showModalAction = () => {
        this.setState({ modalBody: "" });
        this.vengaServiceBusService = serviceBusConnection.getServiceBusService();
        this.generateModalWarningBody().then(bodyResult => this.setState({ modalBody: bodyResult }));//qq needs catch, close modal, needs refactor
    }

    generateModalWarningBody = async () => {
        const vengaServiceBusService = serviceBusConnection.getServiceBusService();
        let numMessages = 0;
        let topicSubList = "";
        let response;

        const messageCountPropertyName = (this.props.messageType === EndpointTypes.MESSAGE) ? 'activeMessageCount' : 'deadletterMessageCount';

        switch (this.props.type) {
            case EndpointTypes.QUEUE:
                response = await vengaServiceBusService.getQueueDetails(this.props.endpointName);
                numMessages = response[messageCountPropertyName];
                break;
            case EndpointTypes.TOPIC:
                response = await vengaServiceBusService.listSubscriptions(this.props.endpointName);
                numMessages = Lodash.sumBy(response, (subscription) => {
                    return subscription[messageCountPropertyName];
                });

                topicSubList = Lodash.map(response, (subscription) => {
                    return subscription.name;
                });
                break;
            case EndpointTypes.SUBSCRIPTION:
                response = await vengaServiceBusService.getSubscriptionDetails(this.props.parentName, this.props.endpointName);
                numMessages = response[messageCountPropertyName];
                break;
            default: break;
        }

        return (
            <React.Fragment>
                <p>Are you sure you want to purge the messages from {this.props.type} "{this.props.endpointName}" ?</p>

                {this.props.type === EndpointTypes.TOPIC ? this.generateTopicModalBody(topicSubList) : ""}

                <p>{numMessages} messages will be deleted <b>irreversibly</b>!</p >
            </React.Fragment>
        );
    }

    generateTopicModalBody = (topicSubList) => {
        const liArray = topicSubList.map(function (listValue, index) {
            return <li key={index}>{listValue}</li>;
        });

        return (
            <React.Fragment>
                <p> This action will purge the messages from the following subscriptions: </p>
                <ul>
                    {liArray}
                </ul>
            </React.Fragment>
        );
    }

    render() {
        const buttonText = this.props.messageType === EndpointTypes.MESSAGE ? "Purge Live Messages" : "Purge Deadletters";
        const buttonSpanText = <span>{buttonText} <Glyphicon glyph="trash" /></span>;

        return (
            <ButtonWithConfirmationModalAndNotification
                id="alertPurge"
                buttonText={buttonSpanText}
                modalTitle={"Purge messages from " + this.props.type}
                modalBody={this.state.modalBody}
                confirmButtonText="Purge"
                afterShowModalAction={this.showModalAction}
                confirmAction={this.getOnPurgeConfirmedHandler()}
                successNotificationMessage="Messages purged successfully!"
                errorNotificationMessage="Messages purging failed!"
                messageAfterSpinner="Purged complete!"
            />
        );
    }
}

export { PurgeMessagesButton };
