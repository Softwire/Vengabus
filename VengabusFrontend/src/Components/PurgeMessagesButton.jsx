import React from 'react';
import { Glyphicon } from 'react-bootstrap';
import { serviceBusConnection } from '../AzureWrappers/ServiceBusConnection';
import { EndpointTypes } from '../Helpers/EndpointTypes';
import Lodash from 'lodash';
import { ButtonWithConfirmationModal } from './ButtonWithConfirmationModal';
import { NotificationManager } from 'react-notifications';

class PurgeMessagesButton extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            modalBody: ""
        };
    }

    getOnPurgeConfirmedHandler = () => {
        const initiatePurge = this.getAPIPurgeAction();
        return () => {
            let purgePromise = initiatePurge();
            //start spinner
            this.setState({ modalBody: "" });
            const showNotificationPromise = purgePromise.then(() => {
                NotificationManager.success("Messages purged successfully!", "Success", 5000);
            }).catch(() => {
                NotificationManager.error("Messages purged failed!", "Error", 999999, undefined, true);
            }).finally(() => {
                this.setState({ modalBody: "Purged finished!" });
                return new Promise((resolve) => {
                    setTimeout(resolve, 2000);
                });
            });
            //return final promise.
            return showNotificationPromise;
        }
    }

    getAPIPurgeAction = () => {
        if (this.props.messageType === EndpointTypes.MESSAGE) {
            switch (this.props.type) {
                case EndpointTypes.TOPIC:
                    return () => this.vengaServiceBusService.purgeTopicMessages(this.props.endpointName);
                case EndpointTypes.QUEUE:
                    return () => this.vengaServiceBusService.purgeQueueMessages(this.props.endpointName);
                case EndpointTypes.SUBSCRIPTION:
                    return () => this.vengaServiceBusService.purgeSubscriptionMessages(this.props.parentName, this.props.endpointName);
                default: break;
            }
        } else {
            switch (this.props.type) {
                case EndpointTypes.TOPIC:
                    return () => this.vengaServiceBusService.purgeTopicDeadletterMessages(this.props.endpointName);
                case EndpointTypes.QUEUE:
                    return () => this.vengaServiceBusService.purgeQueueDeadletterMessages(this.props.endpointName);
                case EndpointTypes.SUBSCRIPTION:
                    return () => this.vengaServiceBusService.purgeSubscriptionDeadletterMessages(this.props.parentName, this.props.endpointName);
                default: break;
            }
        }
    }

    showModalAction = () => {
        this.setState({ modalBody: "" });
        this.vengaServiceBusService = serviceBusConnection.getServiceBusService();
        this.generateModalWarningBody().then(bodyResult => this.setState({ modalBody: bodyResult }));//qq needs catch, close modal, needs refactor
    }

    generateModalWarningBody = async () => {
        const vengaServiceBusService = serviceBusConnection.getServiceBusService();
        let numMessages = 0;
        let topicSubList = "";
        let response;

        if (this.props.messageType === EndpointTypes.MESSAGE) {
            switch (this.props.type) {
                case EndpointTypes.QUEUE:
                    response = await vengaServiceBusService.getQueueDetails(this.props.endpointName);
                    numMessages = response.activeMessageCount;
                    break;
                case EndpointTypes.TOPIC:
                    response = await vengaServiceBusService.listSubscriptions(this.props.endpointName);
                    numMessages = Lodash.sumBy(response, (subscription) => {
                        return subscription.activeMessageCount;
                    });

                    topicSubList = Lodash.map(response, (subscription) => {
                        return subscription.name;
                    });
                    break;
                case EndpointTypes.SUBSCRIPTION:
                    response = await vengaServiceBusService.getSubscriptionDetails(this.props.parentName, this.props.endpointName);
                    numMessages = response.activeMessageCount;
                    break;
                default: break;
            }
        } else {
            switch (this.props.type) {
                case EndpointTypes.QUEUE:
                    response = await vengaServiceBusService.getQueueDetails(this.props.endpointName);
                    numMessages = response.deadletterMessageCount;
                    break;
                case EndpointTypes.TOPIC:
                    response = await vengaServiceBusService.listSubscriptions(this.props.endpointName);
                    numMessages = Lodash.sumBy(response, (subscription) => {
                        return subscription.deadletterMessageCount;
                    });

                    topicSubList = Lodash.map(response, (subscription) => {
                        return subscription.name;
                    });
                    break;
                case EndpointTypes.SUBSCRIPTION:
                    response = await vengaServiceBusService.getSubscriptionDetails(this.props.parentName, this.props.endpointName);
                    numMessages = response.deadletterMessageCount;
                    break;
                default: break;
            }
        }

        return (
            <React.Fragment>
                <p>Are you sure you want to purge the messages from {this.props.type} "{this.props.endpointName}" ?</p>

                {this.props.type === EndpointTypes.TOPIC ? this.generateTopicModalBody(topicSubList) : ""}

                <p>{numMessages} messages will be deleted <b>irreversibly</b>!</p >
            </React.Fragment>
        );
    }

    generateTopicModalBody = (topicSubList) => {
        const liArray = topicSubList.map(function (listValue, index) {
            return <li key={index}>{listValue}</li>;
        });

        return (
            <React.Fragment>
                <p> This action will purge the messages from the following subscriptions: </p>
                <ul>
                    {liArray}
                </ul>
            </React.Fragment>
        );
    }

    render() {
        const buttonText = this.props.messageType === EndpointTypes.MESSAGE
            ? "Purge Live Messages" : "Purge Deadletter Messages";
        const buttonSpanText = <span>{buttonText} <Glyphicon glyph="trash" /></span>;
        return (<ButtonWithConfirmationModal
            id={"alertPurge"}
            buttonText={buttonSpanText}
            modalTitle={"Purge messages from " + this.props.type}
            modalBody={this.state.modalBody}
            confirmButtonText={"Purge"}
            afterShowModalAction={this.showModalAction}
            confirmAction={this.getOnPurgeConfirmedHandler()}
            afterCloseModalAction={() => { }}
        />);
    }
}

export { PurgeMessagesButton };
