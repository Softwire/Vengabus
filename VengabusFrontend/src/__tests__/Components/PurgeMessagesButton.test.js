import { PurgeMessagesButton } from '../../Components/PurgeMessagesButton';
import renderer from 'react-test-renderer';
import React from 'react';
import { mount } from 'enzyme';
import { Modal } from 'react-bootstrap';
import { EndpointTypes } from '../../Helpers/EndpointTypes';
import { testHelper } from '../../TestHelpers/TestHelper';

let mockPurgeSubscrionMessage = jest.fn();
let mockPurgeQueueMessages = jest.fn();
let mockPurgeTopicMessages = jest.fn();
jest.mock('../../AzureWrappers/VengaServiceBusService', () => ({
    VengaServiceBusService: class {
        constructor() {
        }

        getQueueDetails = () => Promise.resolve({
            data: { //qqMDM data!?
                activeMessageCount: 10
            }
        });

        getSubscriptionDetails = () => Promise.resolve({
            data: { //qqMDM data!?
                activeMessageCount: 10
            }
        });

        listSubscriptions = () => Promise.resolve({
            data: [{ //qqMDM data!?
                name: "subscription1",
                activeMessageCount: 10
            }]
        });

        purgeQueueMessages = (queueName) => { mockPurgeQueueMessages(queueName); }
        purgeTopicMessages = (topicName) => { mockPurgeTopicMessages(topicName); }
        purgeSubscriptionMessages = (topicName, subscriptionName) => { mockPurgeSubscrionMessage(subscriptionName); }
    }
}));

function afterModalPurgeButtonIsClicked(wrapper, mockFunction, endpointName) {
    testHelper.clickElementWithId(wrapper, "#alertPurge");


    return testHelper.afterReactHasUpdated().then(() => {
        wrapper.update();
        testHelper.clickElementWithId(wrapper, "#confirm");

        return testHelper.afterReactHasUpdated();

    }).then(() => {
        expect(mockFunction).toHaveBeenCalled();
        expect(mockFunction).toHaveBeenCalledWith(endpointName);
    });
}

describe('PurgeMessagesButton', () => {
    let subscriptionName = "subscriptionName";
    let topicName = "topicName";
    let queueName = "queueName";

    it('renders correctly with given props', () => {
        let deleteMessagesButton = renderer.create(
            <PurgeMessagesButton type={EndpointTypes.SUBSCRIPTION} endpointName={subscriptionName} parentName={topicName} />);
        expect(deleteMessagesButton.toJSON()).toMatchSnapshot();
    });

    it('Modal popup has Purge and Cancel buttons', () => {
        let wrapper = mount(<PurgeMessagesButton type={EndpointTypes.QUEUE} endpointName={queueName} />);
        expect(wrapper.find("#cancel").hostNodes()).toHaveLength(0);
        expect(wrapper.find("#confirm").hostNodes()).toHaveLength(0);
        testHelper.clickElementWithId(wrapper, "#alertPurge");

        return testHelper.afterReactHasUpdated().then(() => {
            wrapper.update();
            expect(wrapper.find("#cancel").hostNodes()).toHaveLength(1);
            expect(wrapper.find("#confirm").hostNodes()).toHaveLength(1);
        });
    });

    it('clicking cancel button does not send purge request to endpoint and closes the Modal', () => {
        let wrapper = mount(<PurgeMessagesButton type={EndpointTypes.QUEUE} endpointName={queueName} />);
        testHelper.clickElementWithId(wrapper, "#alertPurge");

        return testHelper.afterReactHasUpdated().then(() => {
            wrapper.update();
            testHelper.clickElementWithId(wrapper, "#cancel");

            return testHelper.afterReactHasUpdated();
        }).then(() => {
            expect(mockPurgeQueueMessages).not.toHaveBeenCalled();
            expect(wrapper.find(Modal).at(0).prop("show")).toBeFalsy();
        });
    });

    it('call VengaBusService purge subscription messages method once', () => {
        let wrapper = mount(<PurgeMessagesButton type={EndpointTypes.SUBSCRIPTION} endpointName={subscriptionName} parentName={topicName} />);
        return afterModalPurgeButtonIsClicked(wrapper, mockPurgeSubscrionMessage, subscriptionName);
    });

    it('call VengaBusService purge queue messages method once', () => {
        let wrapper = mount(<PurgeMessagesButton type={EndpointTypes.QUEUE} endpointName={queueName} />);
        return afterModalPurgeButtonIsClicked(wrapper, mockPurgeQueueMessages, queueName);
    });

    it('call VengaBusService purge topic messages method once', () => {
        let wrapper = mount(<PurgeMessagesButton type={EndpointTypes.TOPIC} endpointName={topicName} />);
        return afterModalPurgeButtonIsClicked(wrapper, mockPurgeTopicMessages, topicName);
    });
});
