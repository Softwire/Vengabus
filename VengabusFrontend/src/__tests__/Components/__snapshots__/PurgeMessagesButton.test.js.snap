// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurgeMessagesButton renders correctly with given props 1`] = `
<button
  className="btn btn-danger"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  <span>
    Purge Messages 
    <span
      className="glyphicon glyphicon-trash"
    />
  </span>
</button>
`;
